### DISCUSSION TITLE
Create a ThingsDB authentication provider module

### DISCUSSION THREAD

**Discussion by riklempens** [thingsdb/ThingsDB#364] [Ideas]
The idea is to build a ThingsDB module that can be used as an authentication provider by solutions build on ThingsDB.

Supporting OpenID allows us to connect with most common identity providers.

## Reference material

* [What is OpenID Connect](https://openid.net/developers/how-connect-works/)
* [Ory](https://github.com/ory), Open Source Identity Platform For Everyone
* [Run your own OAuth2 Server](https://www.ory.sh/run-oauth2-server-open-source-api-security/)

**Comment by rickmoonex** [comment]
Sound like a great solution for something I'm trying to implement.

Can I be of any help with this?
